### PR TITLE
Dagger DI for `QueryWorkflow`

### LINKED ISSUE #4858: Wire up QueryHandlers to QueryDispatcher

### LINKED ISSUE #4892: Set up Dagger for DI

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Instantiates `QueryWorkflow` via Dagger DI of `QueryWorkflowImpl`.
 - Provides `QueryHandler` access via boilerplate `@Component`'s in each `-service-impl`.
 - Closes #4892 
 - ⚠️ _Note to reviewers_: I don't intend to try to placate Sonar on this one; lmk if you strongly disagree there. 😁 
 - Fixes #4858

### CODE REVIEW

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM as the first step. We can improve on this.

**Review by jeromy-cannon — APPROVED** [review]
release engineering approves the build changes, please wait for other approvers for remaining code before merging.

**Review by kimbor — DISMISSED** [review]

**Review by povolev15 — DISMISSED** [review]

**Review by iwsimon — DISMISSED** [review]
LGTM

**Review by Neeharika-Sompalli — APPROVED** [review]

**Pull request merged**
